src/main/java/karuna/karuna_backend/post/domain/PostService.java

```diff
@@ -21,4 +28,17 @@ public PostDto createPost(PostCreateDto postCreateDto) {
                 .build();
         return PostMapper.toDto(postRepository.save(post));
     }
+
+    private final LoadingCache<Pageable, PostWrapper> cache = Caffeine.newBuilder()
+            .expireAfterWrite(1, TimeUnit.HOURS)
+            .build(this::getPostsFromDatabase);
+
+    public PostWrapper getPosts(Pageable pageable) {
+        return cache.get(pageable);
+    }
+
+    public PostWrapper getPostsFromDatabase(Pageable pageable) {
+        List<Post> posts = postRepository.findAll(pageable).getContent();
```

Comment:
PostWrapper should contain not list, just a page, because there is more details

Reply:
What do you exactly mean by "not list, just a page" if you could elaborate on that one.

Reply:
class Page

Reply:
I corrected it